src/coreclr/nativeaot/Runtime/unix/UnixNativeCodeManager.cpp

```diff
@@ -377,7 +378,78 @@ bool UnixNativeCodeManager::IsUnwindable(PTR_VOID pvAddress)
 #endif
 
     // VirtualUnwind can't unwind epilogues.
-    return TrailingEpilogueInstructionsCount(pMethodInfo, pvAddress) == 0;
+    return TrailingEpilogueInstructionsCount(pMethodInfo, pvAddress) == 0 && IsInProlog(pMethodInfo, pvAddress) != 1;
+}
+
+// checks for known prolog instructions generated by ILC and returns
+//  1 - in prolog
+//  0 - not in prolog,
+// -1 - unknown.
+int UnixNativeCodeManager::IsInProlog(MethodInfo * pMethodInfo, PTR_VOID pvAddress)
+{
+#if defined(TARGET_ARM64)
+
+// post/pre
+
+
+// stp with signed offset
+// x010 1001 00xx xxxx xxxx xxxx xxxx xxxx
+#define STP_BITS1 0x29000000
+#define STP_MASK1 0x7FC00000
+
+// stp with pre/post/no offset
+// x010 100x x0xx xxxx xxxx xxxx xxxx xxxx
+#define STP_BITS2 0x28000000
+#define STP_MASK2 0x7E400000
+
+// add fp, sp, x
+// mov fp, sp
+// 1001 0001 0xxx xxxx xxxx xx11 1111 1101
+#define ADD_FP_SP_BITS 0x910003FD
+#define ADD_FP_SP_MASK 0xFF8003FF
+
+#define STP_RT2_RT_MASK  0x7C1F
+#define STP_RT2_RT_FP_LR 0x781D
+#define STP_RN_MASK      0x3E0
+#define STP_RN_SP        0x3E0
+#define STP_RN_FP        0x3A0
+
+
+    UnixNativeMethodInfo * pNativeMethodInfo = (UnixNativeMethodInfo *)pMethodInfo;
+    ASSERT(pNativeMethodInfo != NULL);
+
+    uint32_t* start  = (uint32_t*)pNativeMethodInfo->pMethodStartAddress;
+    bool savedFpLr = false;
+    bool establishedFp = false;
+
+    for (uint32_t* pInstr = (uint32_t*)start; pInstr < pvAddress && !(savedFpLr && establishedFp); pInstr++)
+    {
+        uint32_t instr = *pInstr;
+
+        if (((instr & STP_MASK1) == STP_BITS1 || (instr & STP_MASK2) == STP_BITS2) &&
+            ((instr & STP_RN_MASK) == STP_RN_SP || (instr & STP_RN_MASK) == STP_RN_FP))
+        {
+            // SP/FP-relative store of pair of registers
+            savedFpLr |= (instr & STP_RT2_RT_MASK) == STP_RT2_RT_FP_LR;
+        }
+        else if ((instr & ADD_FP_SP_MASK) == ADD_FP_SP_BITS)
+        {
+            establishedFp = true;
+        }
+        else
+        {
+            // TODO: Detect saving non-paired register, stack pointer adjustments
```

Comment:
I think it is better for tracking purposes to log a follow-up issue instead of adding a TODO. 

The issue can be triaged/closed/postponed/etc... or described with more info. At least we will not quickly forget about it.

Reply:
It's basically tracked under the umbrella issue https://github.com/dotnet/runtime/issues/76371. I'll update the comments.